### PR TITLE
Save course slides

### PR DESCRIPTION
**Pull request by misteraverin**
This script saves all slides form course in appropriate folders.

I wish Stepic team Merry Christmas and Happy New Year!

### CODE REVIEW

**Comment by misteraverin** [comment]
for running just type: python save_course_slides.py  <id_course>

File: examples/save_course_slides.py [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+
+# get 1 json object
+def fetch_object(obj_class, obj_id):
+    api_url = "{}/{}s/{}".format(api_host, obj_class, obj_id)
+    response = requests.get(api_url, headers={'Authorization': 'Bearer ' + token}).json()
+    return response["{}s".format(obj_class)][0]
+
+
+# get json-objects with ids
+def fetch_objects(obj_class, ids):
```
Comment (mehanig) [inline examples/save_course_slides.py]: This version doesn't not guarantee order of objects, so it's better to use other version.

Pls look for issue - https://github.com/StepicOrg/Stepik-API/pull/53

Whole function could be replaced with new version - https://github.com/StepicOrg/Stepik-API/blob/master/examples/save_course_steps.py#L32 and you will be fine

File: examples/save_course_slides.py [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+
+# create folder for section
+def create_folder(title):
+    if not os.path.exists(os.path.join(CURRENT_PATH, title)):
+        os.makedirs(os.path.join(CURRENT_PATH, title))
+
+
+# download pdf
+def download_file(link, path):
+    slides = urllib.request.urlopen(link)
```
Comment (mehanig) [inline examples/save_course_slides.py]: we already using `requests` , so there is no need to use urllib. `requests` library is preferable way to manage all http requests

**Comment by mehanig** [comment]
it's unobvious the way it's written, why using globals to pass data from function to function.

I like the style other examples written much better. There is no need to put all things in small functions and than manage state of execution using `global` variables declared in many places.

I't might be ok, but is't very differs from other examples.

**Comment by misteraverin** [comment]
@mehanig I fix all marks. What about global variables: I like small functions that easy to use. Because if don't it can become a little bit messy.

**Comment by mehanig** [comment]
Thank you! 
I like small functions too, but when you write 
```
func1()
func2()
```
and data flow from func1 to func2 through global variables, defined inside one of them, it's somehow unclear to understand what's going on.

**Comment by mehanig** [comment]
I didn't get a chance to download any slides, btw.
Am I using it with wrong courses? Tried `67` and `716`

screen shot 2017-01-10 at 7 54 40 pm

screen shot 2017-01-10 at 7 54 17 pm

screen shot 2017-01-10 at 7 59 35 pm

**Comment by misteraverin** [comment]
@mehanig Actually, yes. You choose wrong courses, because not so much courses have slides in attachments. Try 7, 104, 187. I agree with comment about functions.

**Comment by mehanig** [comment]
Got it. Looks good, pls fix prints and add one empty line by yourself, to make commit history clean

Like this https://github.com/StepicOrg/Stepik-API/commit/eec9e93385cb0e927a4f9df646ed041c1d9d14b7 , and I will merge it

Thank you! Great PR 👍

**Comment by misteraverin** [comment]
@mehanig done

**Comment by mehanig** [comment]
👍  merged

**Pull request merged**